include/LLVMSPIRVLib.h

```diff
@@ -106,6 +106,35 @@ std::unique_ptr<SPIRVModule> readSpirvModule(std::istream &IS,
                                              const SPIRV::TranslatorOpts &Opts,
                                              std::string &ErrMsg);
 
+struct SPIRVModuleReport {
+  SPIRV::VersionNumber Version;
+  uint32_t MemoryModel;
+  uint32_t AddrModel;
+  std::vector<std::string> Extensions;
+  std::vector<std::string> ExtensionImports;
```

Comment:
I actually have some questions here:
1. What is the difference between `Extensions` and `ExtensionImports`?
2. The translator defines [extension enum](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/include/LLVMSPIRVExtensions.inc). I believe it would be better to get a vector of enum values instead of strings.

Reply:
1. These are OpExtension and OpExtInst.
2. Thanks, it may be that you are right, I would need a feedback here actually from future customers of the feature. @AlexeySachkov Can you please suggest, maybe numbers indeed make more sense in this binary version of the report that is SPIRVModuleReport?

Reply:
> 1. These are OpExtension and OpExtInst.

Then it probably should be `ExtendedInstructionSets` instead of `ExtensionImports`. Current names look confusing to me.

> 2. Thanks, it may be that you are right, I would need a feedback here actually from future customers of the feature. @AlexeySachkov Can you please suggest, maybe numbers indeed make more sense in this binary version of the report that is SPIRVModuleReport?

As a future customer of this feature, I would say, that number/enum comparison is better for my code, than string comparison.

Reply:
Thank you @vmustya

Reply:
I think that `ExtensionImports` is not properly named indeed. It corresponds to [`OpExtInstImport`](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpExtInstImport) instruction, which imports not an extension, but _extended instruction set_ which is a different mechanism.

> Thanks, it may be that you are right, I would need a feedback here actually from future customers of the feature. @AlexeySachkov Can you please suggest, maybe numbers indeed make more sense in this binary version of the report that is SPIRVModuleReport?

My vision for this new API is that it would be invoked by a SPIR-V consumer before calling `readSpirvModule` as a lightweight and quick operation to parse SPIR-V module "header" to understand if it makes sense (i.e. if it is possible at all) to read and parse the rest of the SPIR-V module.

From that point of view, returning integers/enumerators is better, because it allows to write more performant code. Human-readable names of those extensions are only needed in a negative case when SPIR-V consumer found an unsupported extension simply to prepare a nice error message for a user. The second case is when the tool itself provides a human-readable output in that new `--spirv-print-report` mode, but it should be easy enough to convert enum to a string for the translator.

Reply:
1. I changed names: ExtensionImports into ExtendedInstructionSets, thanks for the suggestion.

2. @vmustya Talking abouts extensions -- I've checked how extensions enum is created and used, and it is not possible to use integer identifier instead of strings for extensions. We are forced to use string names.

Enum that you are referring to does not guarantee that the same extension has the same identifier in two different builds of the library. It is a performance improvement and a way to refer to extensions internally (e.g., for recording users options choice during one run), but not in any case an attempt to assign a unique identifier to an extension: 

https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/9edf71445c624ab4cc6c2213cef917d2e4906e54/include/LLVMSPIRVOpts.h#L71
 
According to spec, extensions are identified by a name, and we are forced to refer to extension using a string name. This is a difference from capabilities, where we should use and will use integers.

Reply:
@VyacheslavLevytskyy, we still can use the enum values. We just shouldn't serialize/save them into the files which can be re-used with a different translator build. For example, we use the enum values to configure SPIR-V writer: https://github.com/intel/cm-compiler/blob/4ba19c8d93f3aa7099ccff2d77038b8a5bd08658/clang/lib/CodeGen/BackendUtil.cpp#L965

Reply:
@VyacheslavLevytskyy, from the other side, we may have a SPIR-V module with a new extension which is not implemented by the translator yet. Obviously we have no enum entries for such extensions. In this case we still should be able to get the list of the extensions from the module. So, the strings might be the only solution.